axon-server-connector/src/main/java/org/axonframework/axonserver/connector/heartbeat/HeartbeatMonitor.java

```diff
@@ -0,0 +1,59 @@
+package org.axonframework.axonserver.connector.heartbeat;
+
+import org.axonframework.axonserver.connector.AxonServerConnectionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Verifies if the connection is still alive, and react if it is not.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2
+ */
+public class HeartbeatMonitor {
+
+    private final Logger log = LoggerFactory.getLogger(HeartbeatMonitor.class);
+
+    private final Runnable onInvalidConnection;
+
+    private final ConnectionSanityCheck connectionSanityCheck;
+
+    /**
+     * Constructs an instance of {@link HeartbeatMonitor} that force a disconnection
+     * when the AxonServer connection is no more alive.
+     *
+     * @param connectionManager connectionManager to AxonServer
+     * @param context           the context
```

Comment:
Nit: Pretty sure we (at AxonIQ) all know what `context` means, but for others this description is a little bit to short/abstract if you ask me.